// @pwdr/tokens/src/mixins/typography.mxn.ts
export const typography = {
    title: {
      xxxl: {
        upper: {
          "font-family":      { value: "pwdr.ref.typography.title.xxxl.font" },
          "font-weight":      { value: "pwdr.ref.typography.title.xxxl.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.title.xxxl.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.title.xxxl.weight" },
          "line-height":      { value: "pwdr.ref.typography.title.xxxl.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.title.xxxl.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        },
        lower: {
          "font-family":      { value: "pwdr.ref.typography.title.xxxl.font" },
          "font-weight":      { value: "pwdr.ref.typography.title.xxxl.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.title.xxxl.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.title.xxxl.weight" },
          "line-height":      { value: "pwdr.ref.typography.title.xxxl.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.title.xxxl.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        }
      },

      xxl: {
        upper: {
          "font-family":      { value: "pwdr.ref.typography.title.xxl.font" },
          "font-weight":      { value: "pwdr.ref.typography.title.xxl.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.title.xxl.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.title.xxl.weight" },
          "line-height":      { value: "pwdr.ref.typography.title.xxl.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.title.xxl.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        },
        lower: {
          "font-family":      { value: "pwdr.ref.typography.title.xxl.font" },
          "font-weight":      { value: "pwdr.ref.typography.title.xxl.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.title.xxl.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.title.xxl.weight" },
          "line-height":      { value: "pwdr.ref.typography.title.xxl.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.title.xxl.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        }
      },

      xl: {
        upper: {
          "font-family":      { value: "pwdr.ref.typography.title.xl.font" },
          "font-weight":      { value: "pwdr.ref.typography.title.xl.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.title.xl.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.title.xl.weight" },
          "line-height":      { value: "pwdr.ref.typography.title.xl.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.title.xl.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        },
        lower: {
          "font-family":      { value: "pwdr.ref.typography.title.xl.font" },
          "font-weight":      { value: "pwdr.ref.typography.title.xl.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.title.xl.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.title.xl.weight" },
          "line-height":      { value: "pwdr.ref.typography.title.xl.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.title.xl.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        }
      },

      lg: {
        upper: {
          "font-family":      { value: "pwdr.ref.typography.title.lg.font" },
          "font-weight":      { value: "pwdr.ref.typography.title.lg.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.title.lg.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.title.lg.weight" },
          "line-height":      { value: "pwdr.ref.typography.title.lg.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.title.lg.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        },
        lower: {
          "font-family":      { value: "pwdr.ref.typography.title.lg.font" },
          "font-weight":      { value: "pwdr.ref.typography.title.lg.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.title.lg.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.title.lg.weight" },
          "line-height":      { value: "pwdr.ref.typography.title.lg.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.title.lg.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        }
      },

      md: {
        upper: {
          "font-family":      { value: "pwdr.ref.typography.title.md.font" },
          "font-weight":      { value: "pwdr.ref.typography.title.md.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.title.md.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.title.md.weight" },
          "line-height":      { value: "pwdr.ref.typography.title.md.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.title.md.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        },
        lower: {
          "font-family":      { value: "pwdr.ref.typography.title.md.font" },
          "font-weight":      { value: "pwdr.ref.typography.title.md.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.title.md.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.title.md.weight" },
          "line-height":      { value: "pwdr.ref.typography.title.md.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.title.md.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        }
      },

      def: {
        upper: {
          "font-family":      { value: "pwdr.ref.typography.title.def.font" },
          "font-weight":      { value: "pwdr.ref.typography.title.def.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.title.def.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.title.def.weight" },
          "line-height":      { value: "pwdr.ref.typography.title.def.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.title.def.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        },
        lower: {
          "font-family":      { value: "pwdr.ref.typography.title.def.font" },
          "font-weight":      { value: "pwdr.ref.typography.title.def.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.title.def.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.title.def.weight" },
          "line-height":      { value: "pwdr.ref.typography.title.def.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.title.def.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        }
      },

      sm: {
        upper: {
          "font-family":      { value: "pwdr.ref.typography.title.sm.font" },
          "font-weight":      { value: "pwdr.ref.typography.title.sm.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.title.sm.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.title.sm.weight" },
          "line-height":      { value: "pwdr.ref.typography.title.sm.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.title.sm.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        },
        lower: {
          "font-family":      { value: "pwdr.ref.typography.title.sm.font" },
          "font-weight":      { value: "pwdr.ref.typography.title.sm.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.title.sm.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.title.sm.weight" },
          "line-height":      { value: "pwdr.ref.typography.title.sm.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.title.sm.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        }
      },

      xs: {
        upper: {
          "font-family":      { value: "pwdr.ref.typography.title.xs.font" },
          "font-weight":      { value: "pwdr.ref.typography.title.xs.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.title.xs.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.title.xs.weight" },
          "line-height":      { value: "pwdr.ref.typography.title.xs.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.title.xs.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        },
        lower: {
          "font-family":      { value: "pwdr.ref.typography.title.xs.font" },
          "font-weight":      { value: "pwdr.ref.typography.title.xs.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.title.xs.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.title.xs.weight" },
          "line-height":      { value: "pwdr.ref.typography.title.xs.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.title.xs.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        }
      },

      xxs: {
        upper: {
          "font-family":      { value: "pwdr.ref.typography.title.xxs.font" },
          "font-weight":      { value: "pwdr.ref.typography.title.xxs.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.title.xxs.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.title.xxs.weight" },
          "line-height":      { value: "pwdr.ref.typography.title.xxs.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.title.xxs.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        },
        lower: {
          "font-family":      { value: "pwdr.ref.typography.title.xxs.font" },
          "font-weight":      { value: "pwdr.ref.typography.title.xxs.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.title.xxs.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.title.xxs.weight" },
          "line-height":      { value: "pwdr.ref.typography.title.xxs.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.title.xxs.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        }
      },

      xxxs: {
        upper: {
          "font-family":      { value: "pwdr.ref.typography.title.xxxs.font" },
          "font-weight":      { value: "pwdr.ref.typography.title.xxxs.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.title.xxxs.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.title.xxxs.weight" },
          "line-height":      { value: "pwdr.ref.typography.title.xxxs.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.title.xxxs.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        },
        lower: {
          "font-family":      { value: "pwdr.ref.typography.title.xxxs.font" },
          "font-weight":      { value: "pwdr.ref.typography.title.xxxs.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.title.xxxs.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.title.xxxs.weight" },
          "line-height":      { value: "pwdr.ref.typography.title.xxxs.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.title.xxxs.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        }
      }
    },

    subtitle: {
      xxxl: {
        upper: {
          "font-family":      { value: "pwdr.ref.typography.subtitle.xxxl.font" },
          "font-weight":      { value: "pwdr.ref.typography.subtitle.xxxl.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.subtitle.xxxl.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.subtitle.xxxl.weight" },
          "line-height":      { value: "pwdr.ref.typography.subtitle.xxxl.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.subtitle.xxxl.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        },
        lower: {
          "font-family":      { value: "pwdr.ref.typography.subtitle.xxxl.font" },
          "font-weight":      { value: "pwdr.ref.typography.subtitle.xxxl.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.subtitle.xxxl.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.subtitle.xxxl.weight" },
          "line-height":      { value: "pwdr.ref.typography.subtitle.xxxl.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.subtitle.xxxl.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        }
      },

      xxl: {
        upper: {
          "font-family":      { value: "pwdr.ref.typography.subtitle.xxl.font" },
          "font-weight":      { value: "pwdr.ref.typography.subtitle.xxl.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.subtitle.xxl.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.subtitle.xxl.weight" },
          "line-height":      { value: "pwdr.ref.typography.subtitle.xxl.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.subtitle.xxl.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        },
        lower: {
          "font-family":      { value: "pwdr.ref.typography.subtitle.xxl.font" },
          "font-weight":      { value: "pwdr.ref.typography.subtitle.xxl.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.subtitle.xxl.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.subtitle.xxl.weight" },
          "line-height":      { value: "pwdr.ref.typography.subtitle.xxl.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.subtitle.xxl.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        }
      },

      xl: {
        upper: {
          "font-family":      { value: "pwdr.ref.typography.subtitle.xl.font" },
          "font-weight":      { value: "pwdr.ref.typography.subtitle.xl.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.subtitle.xl.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.subtitle.xl.weight" },
          "line-height":      { value: "pwdr.ref.typography.subtitle.xl.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.subtitle.xl.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        },
        lower: {
          "font-family":      { value: "pwdr.ref.typography.subtitle.xl.font" },
          "font-weight":      { value: "pwdr.ref.typography.subtitle.xl.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.subtitle.xl.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.subtitle.xl.weight" },
          "line-height":      { value: "pwdr.ref.typography.subtitle.xl.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.subtitle.xl.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        }
      },

      lg: {
        upper: {
          "font-family":      { value: "pwdr.ref.typography.subtitle.lg.font" },
          "font-weight":      { value: "pwdr.ref.typography.subtitle.lg.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.subtitle.lg.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.subtitle.lg.weight" },
          "line-height":      { value: "pwdr.ref.typography.subtitle.lg.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.subtitle.lg.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        },
        lower: {
          "font-family":      { value: "pwdr.ref.typography.subtitle.lg.font" },
          "font-weight":      { value: "pwdr.ref.typography.subtitle.lg.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.subtitle.lg.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.subtitle.lg.weight" },
          "line-height":      { value: "pwdr.ref.typography.subtitle.lg.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.subtitle.lg.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        }
      },

      md: {
        upper: {
          "font-family":      { value: "pwdr.ref.typography.subtitle.md.font" },
          "font-weight":      { value: "pwdr.ref.typography.subtitle.md.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.subtitle.md.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.subtitle.md.weight" },
          "line-height":      { value: "pwdr.ref.typography.subtitle.md.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.subtitle.md.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        },
        lower: {
          "font-family":      { value: "pwdr.ref.typography.subtitle.md.font" },
          "font-weight":      { value: "pwdr.ref.typography.subtitle.md.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.subtitle.md.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.subtitle.md.weight" },
          "line-height":      { value: "pwdr.ref.typography.subtitle.md.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.subtitle.md.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        }
      },

      def: {
        upper: {
          "font-family":      { value: "pwdr.ref.typography.subtitle.def.font" },
          "font-weight":      { value: "pwdr.ref.typography.subtitle.def.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.subtitle.def.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.subtitle.def.weight" },
          "line-height":      { value: "pwdr.ref.typography.subtitle.def.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.subtitle.def.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        },
        lower: {
          "font-family":      { value: "pwdr.ref.typography.subtitle.def.font" },
          "font-weight":      { value: "pwdr.ref.typography.subtitle.def.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.subtitle.def.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.subtitle.def.weight" },
          "line-height":      { value: "pwdr.ref.typography.subtitle.def.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.subtitle.def.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        }
      },

      sm: {
        upper: {
          "font-family":      { value: "pwdr.ref.typography.subtitle.sm.font" },
          "font-weight":      { value: "pwdr.ref.typography.subtitle.sm.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.subtitle.sm.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.subtitle.sm.weight" },
          "line-height":      { value: "pwdr.ref.typography.subtitle.sm.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.subtitle.sm.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        },
        lower: {
          "font-family":      { value: "pwdr.ref.typography.subtitle.sm.font" },
          "font-weight":      { value: "pwdr.ref.typography.subtitle.sm.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.subtitle.sm.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.subtitle.sm.weight" },
          "line-height":      { value: "pwdr.ref.typography.subtitle.sm.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.subtitle.sm.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        }
      },

      xs: {
        upper: {
          "font-family":      { value: "pwdr.ref.typography.subtitle.xs.font" },
          "font-weight":      { value: "pwdr.ref.typography.subtitle.xs.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.subtitle.xs.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.subtitle.xs.weight" },
          "line-height":      { value: "pwdr.ref.typography.subtitle.xs.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.subtitle.xs.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        },
        lower: {
          "font-family":      { value: "pwdr.ref.typography.subtitle.xs.font" },
          "font-weight":      { value: "pwdr.ref.typography.subtitle.xs.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.subtitle.xs.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.subtitle.xs.weight" },
          "line-height":      { value: "pwdr.ref.typography.subtitle.xs.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.subtitle.xs.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        }
      },

      xxs: {
        upper: {
          "font-family":      { value: "pwdr.ref.typography.subtitle.xxs.font" },
          "font-weight":      { value: "pwdr.ref.typography.subtitle.xxs.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.subtitle.xxs.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.subtitle.xxs.weight" },
          "line-height":      { value: "pwdr.ref.typography.subtitle.xxs.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.subtitle.xxs.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        },
        lower: {
          "font-family":      { value: "pwdr.ref.typography.subtitle.xxs.font" },
          "font-weight":      { value: "pwdr.ref.typography.subtitle.xxs.weight-bold" },
          "font-size":        { value: "pwdr.ref.typography.subtitle.xxs.size" },
          "font-weight-bold": { value: "pwdr.ref.typography.subtitle.xxs.weight" },
          "line-height":      { value: "pwdr.ref.typography.subtitle.xxs.line-hieght" },
          "letter-spacing":   { value: "pwdr.ref.typography.subtitle.xxs.tracking" },
          "text-transform":   { value: "pwdr.ref.typography.case.upper" }
        }
      }
    },

    body: {
      lg: {
        "font-family":      { value: "pwdr.ref.typography.body.lg.font" },
        "font-weight":      { value: "pwdr.ref.typography.body.lg.weight-bold" },
        "font-size":        { value: "pwdr.ref.typography.body.lg.size" },
        "font-weight-bold": { value: "pwdr.ref.typography.body.lg.weight" },
        "line-height":      { value: "pwdr.ref.typography.body.lg.line-hieght" },
        "letter-spacing":   { value: "pwdr.ref.typography.body.lg.tracking" }
        // no Case in CSV → no text-transform
      },
      md: {
        "font-family":      { value: "pwdr.ref.typography.body.md.font" },
        "font-weight":      { value: "pwdr.ref.typography.body.md.weight-bold" },
        "font-size":        { value: "pwdr.ref.typography.body.md.size" },
        "font-weight-bold": { value: "pwdr.ref.typography.body.md.weight" },
        "line-height":      { value: "pwdr.ref.typography.body.md.line-hieght" },
        "letter-spacing":   { value: "pwdr.ref.typography.body.md.tracking" }
      },
      sm: {
        "font-family":      { value: "pwdr.ref.typography.body.sm.font" },
        "font-weight":      { value: "pwdr.ref.typography.body.sm.weight-bold" },
        "font-size":        { value: "pwdr.ref.typography.body.sm.size" },
        "font-weight-bold": { value: "pwdr.ref.typography.body.sm.weight" },
        "line-height":      { value: "pwdr.ref.typography.body.sm.line-hieght" },
        "letter-spacing":   { value: "pwdr.ref.typography.body.sm.tracking" }
      }
    }
} as const;

export type TypographyMixins = typeof typography;
