// @pwdr/tokens/src/deprecated/typography.ref.ts
export const typography = {
    title: {
      xxxl: {
        font: {
          value: "pwdr.sys.typography.font-family"
        },
        "weight-bold": {
          value: "pwdr.sys.typography.font-weight.bold"
        },
        size: {
          value: "pwdr.sys.typography.font-size.64"
        },
        weight: {
          value: "pwdr.sys.typography.font-weight.semibold"
        },
        "line-height": {
          value: "none"
        },
        tracking: {
          value: "none"
        }
      },
      xxl: {
        font: {
          value: "pwdr.sys.typography.font-family"
        },
        "weight-bold": {
          value: "pwdr.sys.typography.font-weight.bold"
        },
        size: {
          value: "pwdr.sys.typography.font-size.48"
        },
        weight: {
          value: "pwdr.sys.typography.font-weight.semibold"
        },
        "line-height": {
          value: "none"
        },
        tracking: {
          value: "none"
        }
      },
      xl: {
        font: {
          value: "pwdr.sys.typography.font-family"
        },
        "weight-bold": {
          value: "pwdr.sys.typography.font-weight.bold"
        },
        size: {
          value: "pwdr.sys.typography.font-size.32"
        },
        weight: {
          value: "pwdr.sys.typography.font-weight.semibold"
        },
        "line-height": {
          value: "none"
        },
        tracking: {
          value: "none"
        }
      },
      lg: {
        font: {
          value: "pwdr.sys.typography.font-family"
        },
        "weight-bold": {
          value: "pwdr.sys.typography.font-weight.bold"
        },
        size: {
          value: "pwdr.sys.typography.font-size.24"
        },
        weight: {
          value: "pwdr.sys.typography.font-weight.semibold"
        },
        "line-height": {
          value: "none"
        },
        tracking: {
          value: "none"
        }
      },
      md: {
        font: {
          value: "pwdr.sys.typography.font-family"
        },
        "weight-bold": {
          value: "pwdr.sys.typography.font-weight.bold"
        },
        size: {
          value: "pwdr.sys.typography.font-size.20"
        },
        weight: {
          value: "pwdr.sys.typography.font-weight.semibold"
        },
        "line-height": {
          value: "none"
        },
        tracking: {
          value: "none"
        }
      },
      // plain "title" size (no suffix)
      font: {
        value: "pwdr.sys.typography.font-family"
      },
      "weight-bold": {
        value: "pwdr.sys.typography.font-weight.bold"
      },
      size: {
        value: "pwdr.sys.typography.font-size.16"
      },
      weight: {
        value: "pwdr.sys.typography.font-weight.semibold"
      },
      "line-height": {
        value: "none"
      },
      tracking: {
        value: "none"
      },
      sm: {
        font: {
          value: "pwdr.sys.typography.font-family"
        },
        "weight-bold": {
          value: "pwdr.sys.typography.font-weight.bold"
        },
        size: {
          value: "pwdr.sys.typography.font-size.14"
        },
        weight: {
          value: "pwdr.sys.typography.font-weight.semibold"
        },
        "line-height": {
          value: "none"
        },
        tracking: {
          value: "none"
        }
      },
      xs: {
        font: {
          value: "pwdr.sys.typography.font-family"
        },
        "weight-bold": {
          value: "pwdr.sys.typography.font-weight.bold"
        },
        size: {
          value: "pwdr.sys.typography.font-size.12"
        },
        weight: {
          value: "pwdr.sys.typography.font-weight.semibold"
        },
        "line-height": {
          value: "none"
        },
        tracking: {
          value: "none"
        }
      },
      xxs: {
        font: {
          value: "pwdr.sys.typography.font-family"
        },
        "weight-bold": {
          value: "pwdr.sys.typography.font-weight.bold"
        },
        size: {
          value: "pwdr.sys.typography.font-size.10"
        },
        weight: {
          value: "pwdr.sys.typography.font-weight.semibold"
        },
        "line-height": {
          value: "none"
        },
        tracking: {
          value: "none"
        }
      },
      xxxs: {
        font: {
          value: "pwdr.sys.typography.font-family"
        },
        "weight-bold": {
          value: "pwdr.sys.typography.font-weight.bold"
        },
        size: {
          value: "pwdr.sys.typography.font-size.8"
        },
        weight: {
          value: "pwdr.sys.typography.font-weight.semibold"
        },
        "line-height": {
          value: "none"
        },
        tracking: {
          value: "none"
        }
      }
    },

    subtitle: {
      xxxl: {
        font: {
          value: "pwdr.sys.typography.font-family"
        },
        "weight-bold": {
          value: "pwdr.sys.typography.font-weight.bold"
        },
        size: {
          value: "pwdr.sys.typography.font-size.48"
        },
        weight: {
          value: "pwdr.sys.typography.font-weight.medium"
        },
        "line-height": {
          value: "none"
        },
        tracking: {
          value: "none"
        }
      },
      xxl: {
        font: {
          value: "pwdr.sys.typography.font-family"
        },
        "weight-bold": {
          value: "pwdr.sys.typography.font-weight.bold"
        },
        size: {
          value: "pwdr.sys.typography.font-size.32"
        },
        weight: {
          value: "pwdr.sys.typography.font-weight.medium"
        },
        "line-height": {
          value: "none"
        },
        tracking: {
          value: "none"
        }
      },
      xl: {
        font: {
          value: "pwdr.sys.typography.font-family"
        },
        "weight-bold": {
          value: "pwdr.sys.typography.font-weight.bold"
        },
        size: {
          value: "pwdr.sys.typography.font-size.24"
        },
        weight: {
          value: "pwdr.sys.typography.font-weight.medium"
        },
        "line-height": {
          value: "none"
        },
        tracking: {
          value: "none"
        }
      },
      lg: {
        font: {
          value: "pwdr.sys.typography.font-family"
        },
        "weight-bold": {
          value: "pwdr.sys.typography.font-weight.bold"
        },
        size: {
          value: "pwdr.sys.typography.font-size.20"
        },
        weight: {
          value: "pwdr.sys.typography.font-weight.medium"
        },
        "line-height": {
          value: "none"
        },
        tracking: {
          value: "none"
        }
      },
      md: {
        font: {
          value: "pwdr.sys.typography.font-family"
        },
        "weight-bold": {
          value: "pwdr.sys.typography.font-weight.bold"
        },
        size: {
          value: "pwdr.sys.typography.font-size.16"
        },
        weight: {
          value: "pwdr.sys.typography.font-weight.medium"
        },
        "line-height": {
          value: "none"
        },
        tracking: {
          value: "none"
        }
      },
      // plain "subtitle" size (no suffix)
      font: {
        value: "pwdr.sys.typography.font-family"
      },
      "weight-bold": {
        value: "pwdr.sys.typography.font-weight.bold"
      },
      size: {
        value: "pwdr.sys.typography.font-size.14"
      },
      // note: CSV has a ref here: pwdr.ref.typography.font-weight.medium
      // keeping it as-is so your recursive resolver can handle it if/when you define it.
      weight: {
        value: "pwdr.ref.typography.font-weight.medium"
      },
      "line-height": {
        value: "none"
      },
      tracking: {
        value: "none"
      },
      sm: {
        font: {
          value: "pwdr.sys.typography.font-family"
        },
        "weight-bold": {
          value: "pwdr.sys.typography.font-weight.bold"
        },
        size: {
          value: "pwdr.sys.typography.font-size.12"
        },
        weight: {
          value: "pwdr.sys.typography.font-weight.medium"
        },
        "line-height": {
          value: "none"
        },
        tracking: {
          value: "none"
        }
      },
      xs: {
        font: {
          value: "pwdr.sys.typography.font-family"
        },
        "weight-bold": {
          value: "pwdr.sys.typography.font-weight.bold"
        },
        size: {
          value: "pwdr.sys.typography.font-size.10"
        },
        weight: {
          value: "pwdr.sys.typography.font-weight.medium"
        },
        "line-height": {
          value: "none"
        },
        tracking: {
          value: "none"
        }
      },
      xxs: {
        font: {
          value: "pwdr.sys.typography.font-family"
        },
        "weight-bold": {
          value: "pwdr.sys.typography.font-weight.bold"
        },
        size: {
          value: "pwdr.sys.typography.font-size.8"
        },
        weight: {
          value: "pwdr.sys.typography.font-weight.medium"
        },
        "line-height": {
          value: "none"
        },
        tracking: {
          value: "none"
        }
      }
    },

    body: {
      lg: {
        font: {
          value: "pwdr.sys.typography.font-family"
        },
        "weight-bold": {
          value: "pwdr.sys.typography.font-weight.bold"
        },
        size: {
          value: "pwdr.sys.typography.font-size.14"
        },
        weight: {
          value: "pwdr.sys.typography.font-weight.regular"
        },
        "line-height": {
          value: "pwdr.sys.typography.line-hieght.md"
        },
        tracking: {
          value: "none"
        }
      },
      md: {
        font: {
          value: "pwdr.sys.typography.font-family"
        },
        "weight-bold": {
          value: "pwdr.sys.typography.font-weight.bold"
        },
        size: {
          value: "pwdr.sys.typography.font-size.12"
        },
        weight: {
          value: "pwdr.sys.typography.font-weight.regular"
        },
        "line-height": {
          value: "pwdr.sys.typography.line-hieght.md"
        },
        tracking: {
          value: "none"
        }
      },
      sm: {
        font: {
          value: "pwdr.sys.typography.font-family"
        },
        "weight-bold": {
          value: "pwdr.sys.typography.font-weight.bold"
        },
        size: {
          value: "pwdr.sys.typography.font-size.10"
        },
        weight: {
          value: "pwdr.sys.typography.font-weight.regular"
        },
        "line-height": {
          value: "pwdr.sys.typography.line-hieght.md"
        },
        tracking: {
          value: "none"
        }
      }
    }
} as const;

export type TypographyReference = typeof typography;
