// @pwdr/tokens/src/reference/color.ref.ts
export const color = {
    border: {
      default: {
        value: "pwdr.sys.color.palette.neutral.050",
        dark: { value: "None" }
      },
      secondary: {
        value: "pwdr.sys.color.palette.neutral.000",
        dark: { value: "None" }
      },
      neutral: {
        value: "pwdr.sys.color.palette.neutral.500",
        dark: { value: "None" }
      },
      active: {
        value: "pwdr.sys.color.palette.primary.500",
        dark: { value: "None" }
      },
      bold: {
        value: "pwdr.sys.color.palette.neutral.950",
        dark: { value: "None" }
      },
      alert: {
        value: "pwdr.sys.color.palette.red.500",
        dark: { value: "None" }
      },
      warning: {
        value: "pwdr.sys.color.palette.yellow.500",
        dark: { value: "None" }
      },
      success: {
        value: "pwdr.ref.color.palette.c05-00",
        dark: { value: "None" }
      },
      diabled: {
        value: "pwdr.sys.color.palette.grey.500",
        dark: { value: "None" }
      },
      asset: {
        value: "pwdr.sys.color.palette.periwinkle.500",
        dark: { value: "None" }
      },
      discovery: {
        value: "pwdr.sys.color.palette.purple.500",
        dark: { value: "None" }
      }
    },

    shadow: {
      default: {
        value: "pwdr.sys.color.palette.neutral.500",
        dark: { value: "None" }
      },
      active: {
        value: "pwdr.sys.color.palette.primary.500",
        dark: { value: "None" }
      },
      bold: {
        value: "pwdr.sys.color.palette.neutral.950",
        dark: { value: "None" }
      },
      alert: {
        value: "pwdr.sys.color.palette.red.500",
        dark: { value: "None" }
      },
      warning: {
        value: "pwdr.sys.color.palette.yellow.500",
        dark: { value: "None" }
      },
      success: {
        value: "pwdr.sys.color.palette.green.500",
        dark: { value: "None" }
      },
      diabled: {
        value: "pwdr.sys.color.palette.grey.500",
        dark: { value: "None" }
      },
      asset: {
        value: "pwdr.sys.color.palette.periwinkle.500",
        dark: { value: "None" }
      },
      discovery: {
        value: "pwdr.sys.color.palette.purple.500",
        dark: { value: "None" }
      }
    },

    state: {
      default: {
        value: "pwdr.sys.color.palette.neutral.000",
        dark: { value: "None" }
      },
      secondary: {
        value: "pwdr.sys.color.palette.neutral.050",
        dark: { value: "None" }
      },
      neutral: {
        value: "pwdr.sys.color.palette.neutral.500",
        dark: { value: "None" }
      },
      active: {
        value: "pwdr.sys.color.palette.primary.500",
        dark: { value: "None" }
      },
      bold: {
        value: "pwdr.sys.color.palette.neutral.950",
        dark: { value: "None" }
      },
      alert: {
        value: "pwdr.sys.color.palette.red.500",
        dark: { value: "None" }
      },
      warning: {
        value: "pwdr.sys.color.palette.yellow.500",
        dark: { value: "None" }
      },
      success: {
        value: "pwdr.sys.color.palette.green.500",
        dark: { value: "None" }
      },
      diabled: {
        value: "pwdr.sys.color.palette.grey.500",
        dark: { value: "None" }
      },
      asset: {
        value: "pwdr.sys.color.palette.periwinkle.500",
        dark: { value: "None" }
      },
      discovery: {
        value: "pwdr.sys.color.palette.purple.500",
        dark: { value: "None" }
      }
    },

    icon: {
      default: {
        value: "pwdr.sys.color.palette.neutral.000",
        dark: { value: "None" }
      },
      secondary: {
        value: "pwdr.sys.color.palette.neutral.050",
        dark: { value: "None" }
      },
      neutral: {
        value: "pwdr.sys.color.palette.neutral.500",
        dark: { value: "None" }
      },
      active: {
        value: "pwdr.sys.color.palette.primary.500",
        dark: { value: "None" }
      },
      bold: {
        value: "pwdr.sys.color.palette.neutral.950",
        dark: { value: "None" }
      },
      alert: {
        value: "pwdr.sys.color.palette.red.500",
        dark: { value: "None" }
      },
      warning: {
        value: "pwdr.sys.color.palette.yellow.500",
        dark: { value: "None" }
      },
      success: {
        value: "pwdr.sys.color.palette.green.500",
        dark: { value: "None" }
      },
      diabled: {
        value: "pwdr.sys.color.palette.grey.500",
        dark: { value: "None" }
      },
      asset: {
        value: "pwdr.sys.color.palette.periwinkle.500",
        dark: { value: "None" }
      },
      discovery: {
        value: "pwdr.sys.color.palette.purple.500",
        dark: { value: "None" }
      }
    },

    surface: {
      default: {
        value: "pwdr.sys.color.palette.neutral.000",
        dark: { value: "None" }
      },
      "default.seconday": {
        value: "pwdr.sys.color.palette.primary.200",
        dark: { value: "None" }
      },
      "default.tertariy": {
        value: "pwdr.sys.color.palette.primary.100",
        dark: { value: "None" }
      },
      "default.hover": {
        value: "pwdr.sys.color.palette.neutral.050",
        dark: { value: "None" }
      },
      neutral: {
        value: "pwdr.sys.color.palette.neutral.050",
        dark: { value: "None" }
      },
      "neutral.hover": {
        value: "None",
        dark: { value: "None" }
      },
      active: {
        value: "pwdr.sys.color.palette.primary.500",
        dark: { value: "None" }
      },
      "active.hover": {
        value: "pwdr.sys.color.palette.primary.300",
        dark: { value: "None" }
      },
      bold: {
        value: "pwdr.sys.color.palette.neutral.950",
        dark: { value: "None" }
      },
      "bold.hover": {
        value: "pwdr.sys.color.palette.neutral.600",
        dark: { value: "None" }
      },
      alert: {
        value: "pwdr.sys.color.palette.red.500",
        dark: { value: "None" }
      },
      warning: {
        value: "pwdr.sys.color.palette.yellow.500",
        dark: { value: "None" }
      },
      success: {
        value: "pwdr.sys.color.palette.green.500",
        dark: { value: "None" }
      },
      diabled: {
        value: "pwdr.sys.color.palette.grey.500",
        dark: { value: "None" }
      },
      asset: {
        value: "pwdr.sys.color.palette.periwinkle.500",
        dark: { value: "None" }
      },
      discovery: {
        value: "pwdr.sys.color.palette.purple.500",
        dark: { value: "None" }
      },
      "alert.hover": {
        value: "None",
        dark: { value: "None" }
      },
      "warning.hover": {
        value: "None",
        dark: { value: "None" }
      },
      "success.hover": {
        value: "None",
        dark: { value: "None" }
      },
      "diabled.hover": {
        value: "None",
        dark: { value: "None" }
      },
      "asset.hover": {
        value: "None",
        dark: { value: "None" }
      },
      "discovery.hover": {
        value: "None",
        dark: { value: "None" }
      }
    },

    text: {
      "on-surface-default": {
        value: "pwdr.sys.color.palette.neutral.950",
        dark: { value: "None" }
      },
      "on-surface-default.secondary": {
        value: "pwdr.sys.color.palette.neutral.950",
        dark: { value: "None" }
      },
      "on-surface-default.tertariy": {
        value: "pwdr.sys.color.palette.neutral.950",
        dark: { value: "None" }
      },
      "on-surface-default.hover": {
        value: "pwdr.sys.color.palette.neutral.950",
        dark: { value: "None" }
      },
      "on-surface-neutral": {
        value: "pwdr.sys.color.palette.neutral.950",
        dark: { value: "None" }
      },
      "on-surface-neutral.hover": {
        value: "pwdr.sys.color.palette.neutral.950",
        dark: { value: "None" }
      },
      "on-surface-active": {
        value: "pwdr.sys.color.palette.neutral.000",
        dark: { value: "None" }
      },
      "on-surface-active.hover": {
        value: "pwdr.sys.color.palette.neutral.000",
        dark: { value: "None" }
      },
      "on-surface-bold": {
        value: "pwdr.sys.color.palette.neutral.000",
        dark: { value: "None" }
      },
      "on-surface-bold.hover": {
        value: "pwdr.sys.color.palette.neutral.000",
        dark: { value: "None" }
      },
      "on-surface-alert": {
        value: "pwdr.sys.color.palette.neutral.000",
        dark: { value: "None" }
      },
      "on-surface-warning": {
        value: "pwdr.sys.color.palette.neutral.000",
        dark: { value: "None" }
      },
      "on-surface-success": {
        value: "pwdr.sys.color.palette.neutral.000",
        dark: { value: "None" }
      },
      "on-surface-diabled": {
        value: "pwdr.sys.color.palette.neutral.500",
        dark: { value: "None" }
      },
      "on-surface-asset": {
        value: "pwdr.sys.color.palette.neutral.950",
        dark: { value: "None" }
      },
      "on-surface-discovery": {
        value: "pwdr.sys.color.palette.neutral.000",
        dark: { value: "None" }
      },
      "on-surface-alert.hover": {
        value: "pwdr.sys.color.palette.neutral.000",
        dark: { value: "None" }
      },
      "on-surface-warning.hover": {
        value: "pwdr.sys.color.palette.neutral.000",
        dark: { value: "None" }
      },
      "on-surface-success.hover": {
        value: "pwdr.sys.color.palette.neutral.000",
        dark: { value: "None" }
      },
      "on-surface-diabled.hover": {
        value: "pwdr.sys.color.palette.neutral.500",
        dark: { value: "None" }
      },
      "on-surface-asset.hover": {
        value: "pwdr.sys.color.palette.neutral.950",
        dark: { value: "None" }
      },
      "on-surface-discovery.hover": {
        value: "pwdr.sys.color.palette.neutral.000",
        dark: { value: "None" }
      },
      default: {
        value: "pwdr.sys.color.palette.neutral.950",
        dark: { value: "None" }
      },
      "default.seconday": {
        value: "pwdr.sys.color.palette.primary.300",
        dark: { value: "None" }
      },
      neutral: {
        value: "pwdr.sys.color.palette.neutral.500",
        dark: { value: "None" }
      },
      active: {
        value: "pwdr.sys.color.palette.primary.500",
        dark: { value: "None" }
      },
      inverse: {
        value: "pwdr.sys.color.palette.neutral.000",
        dark: { value: "None" }
      },
      alert: {
        value: "pwdr.sys.color.palette.red.500",
        dark: { value: "None" }
      },
      warning: {
        value: "pwdr.sys.color.palette.yellow.500",
        dark: { value: "None" }
      },
      success: {
        value: "pwdr.sys.color.palette.green.500",
        dark: { value: "None" }
      },
      diabled: {
        value: "pwdr.sys.color.palette.grey.500",
        dark: { value: "None" }
      },
      asset: {
        value: "pwdr.sys.color.palette.periwinkle.500",
        dark: { value: "None" }
      },
      discovery: {
        value: "pwdr.sys.color.palette.purple.500",
        dark: { value: "None" }
      }
    }
  } as const;

export type ColorReference = typeof color;
